src/main/kotlin/io/github/inductiveautomation/kindling/thread/comparison/DetailContainer.kt

```diff
@@ -89,12 +80,35 @@ internal class DetailContainer(
         }
 
     init {
-        add(headerButton, "growx, top, wmax 100%")
-        add(scrollPane, "push, grow, top, wmax 100%")
+        add(headerButton, "growx, top")
+        add(scrollPane, "push, grow, top")
     }
 
     companion object {
         private val collapseIcon = FlatSVGIcon("icons/bx-chevrons-up.svg")
         private val expandIcon = FlatSVGIcon("icons/bx-chevrons-down.svg")
     }
+
+    inner class HeaderButton : JButton() {
+        val expandLabel = JLabel(expandIcon).apply { isVisible = isCollapsed }
+        val collapseLabel = JLabel(collapseIcon).apply { isVisible = !isCollapsed }
+        val textLabel = JLabel(prefix).apply {
```

Comment:
Minor: These could be private and you could override `getText/setText` from the base JButton to forward into the inner `textLabel`, so you're not "leaking" implementation details of your layout class.

Reply:
@paul-griffith I actually tried that yesterday but ran into a weird issue. I'll try it again quickly to see if it was something obvious I missed